prompt_prototype.rb

```diff
@@ -18,35 +19,22 @@
 ep_country_and_house, morph_scraper, ep_id_scheme, wikidata_membership_item, scraper_sql = ARGV
 scraper_sql ||= 'SELECT * FROM data'
 
-unless ENV['MORPH_API_KEY']
-  abort "You must set MORPH_API_KEY in the environment"
-end
+morph_list = MembershipLists::Morph.new(
+  morph_scraper: morph_scraper,
+  morph_sql_query: scraper_sql
+)
 
-WIKIDATA_SPARQL_URL = 'https://query.wikidata.org/sparql'
+morph_records = morph_list.to_a.map { |d| [d[:id], d] }.to_h
 
-def wikidata_sparql(query)
-  result = RestClient.get WIKIDATA_SPARQL_URL, params: { query: query, format: 'json' }
-  json = JSON.parse(result, symbolize_names: true)
-  json[:results][:bindings].map do |res|
-    url = res[:item][:value]
-    item_id = url.split('/').last
-    [item_id, {url: url, name: res[:itemLabel][:value]}]
-  end.to_h
-rescue RestClient::Exception => e
-  abort "Wikidata query #{query.inspect} failed: #{e.message}"
-end
-
-def morph_records(scraper, query)
-  url = "https://morph.io/#{scraper}/data.json?key=#{ENV['MORPH_API_KEY']}&query=#{URI.encode_www_form_component(query)}"
-  JSON.parse(RestClient.get(url), symbolize_names: true).map { |d| [d[:id], d]  }.to_h
-end
+wikidata_list = MembershipLists::Wikidata.new(
+  wikidata_membership_item: wikidata_membership_item
+)
 
-wikidata_records = wikidata_sparql("SELECT ?item ?itemLabel WHERE { ?item wdt:P39 wd:#{wikidata_membership_item}.  SERVICE wikibase:label { bd:serviceParam wikibase:language \"[AUTO_LANGUAGE],en\". } }")
-morph_records = morph_records(morph_scraper, scraper_sql)
+wikidata_records = wikidata_list.to_a.map { |h| [h[:item_id], h] }.to_h
```

Comment:
In both cases here you're immediately post-processing the `to_a` call to convert the result to a hash. Is this just an interim approach, or does it imply that it might be useful to have a method like that on the MembershipList classes?